### ISSUE TITLE
IPv6 support on LoadBalancerSourceRanges 

### ISSUE THREAD

**Issue by aristosvo** [Azure/AKS#1614]
**What happened**:
When I configure an IPv6 address to be whitelisted in my Network Security Group by setting it in loadBalancerSourceRanges in my Ingress Service:
```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2020-04-17T18:56:35Z"
  finalizers:
  - service.kubernetes.io/load-balancer-cleanup
  labels:
    app: nginx-ingress
    chart: nginx-ingress-1.24.4
    component: controller
    heritage: Helm
    release: ingress
  name: ingress-nginx-ingress-controller
  namespace: ingress
  resourceVersion: "8618804"
  selfLink: /api/v1/namespaces/ingress/services/ingress-nginx-ingress-controller
  uid: 4b942d4b-6310-45fa-881c-dcac40f7990b
spec:
  clusterIP: xx.x.xx.xxx
  externalTrafficPolicy: Local
  healthCheckNodePort: 31095
  loadBalancerSourceRanges:
  - xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx/128
  ports:
  - name: http
    nodePort: 32687
    port: 80
    protocol: TCP
    targetPort: http
  - name: https
    nodePort: 31014
    port: 443
    protocol: TCP
    targetPort: https
  selector:
    app: nginx-ingress
    component: controller
    release: ingress
  sessionAffinity: None
  type: LoadBalancer
```

I get these events/errors:
```
+ kubectl get events
LAST SEEN   TYPE      REASON                        OBJECT                                     MESSAGE
20m         Normal    EnsuringLoadBalancer          service/ingress-nginx-ingress-controller   Ensuring load balancer
20m         Normal    EnsuredLoadBalancer           service/ingress-nginx-ingress-controller   Ensured load balancer
12m         Warning   KubeProxyIncorrectIPVersion   service/ingress-nginx-ingress-controller   xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx/128 in loadBalancerSourceRanges has incorrect IP version
12m         Warning   KubeProxyIncorrectIPVersion   service/ingress-nginx-ingress-controller   xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx/128 in loadBalancerSourceRanges has incorrect IP version
20m         Normal    LoadBalancerSourceRanges      service/ingress-nginx-ingress-controller    xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx/128]
12m         Warning   KubeProxyIncorrectIPVersion   service/ingress-nginx-ingress-controller   xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx/128 in loadBalancerSourceRanges has incorrect IP version
12m         Warning   KubeProxyIncorrectIPVersion   service/ingress-nginx-ingress-controller   xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx/128 in loadBalancerSourceRanges has incorrect IP version
20m         Warning   CreateOrUpdateSecurityGroup   service/ingress-nginx-ingress-controller   Code="InvalidResourceName" Message="Resource name xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-TCP-80-xxxx:xxxx:xxxx:xxxx:xxxx:xxx:xxxx:xxxx_128 is invalid. The name can be up to 80 characters long. It must begin with a word character, and it must end with a word character or with '_'. The name may contain word characters or '.', '-', '_'." Details=[]
20m         Warning   SyncLoadBalancerFailed        service/ingress-nginx-ingress-controller   Error syncing load balancer: failed to ensure load balancer: timed out waiting for the condition
```

**What you expected to happen**:
Configuration on Network Security Group for whitelisted IP's to be updated

**How to reproduce it (as minimally and precisely as possible)**:
- `helm repo add add stable https://kubernetes-charts.storage.googleapis.com/`
- `helm install ingress stable/ingress --version 0.26.1 -n ingress`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
v1.16.7
- Size of cluster (how many worker nodes are in the cluster?)
3

**Comment by glothriel** [comment]
@aristosvo Did you manage to configure a Service of type LoadBalancer, that is listening on ipv6 address? If so, how did you do this? Is there any specific setup around azure network primitives required to make this working?

**Comment by phealy** [comment]
Will be addressed as part of #460

**Comment by wangyira** [comment]
IPv6 Kubenet support is in public preview and available in all public regions: https://aka.ms/aks/ipv6
